prototype/docs/southbound/device-factory/tuto-parser.md

```diff
@@ -0,0 +1,367 @@
+# Tutorial: How to write a custom Device Factory Parser
+
+It is possible and quite easy to create your own device factory parser.
+
+First, choose an available parser ID: it will be used in a JSON configuration file and in a service property so you might want to avoid non-ASCII characters.
+In this tutorial, we will consider the parser ID to be `tuto-custom-parser`.
+Then, define a device mapping record type implementing the `org.eclipse.sensinact.gateway.southbound.device.factory.IDeviceMappingRecord` interface.
+Finally, design your parser keeping in mind that:
+
+* a parser is considered a singleton: it will be called each time a payload associated to that parser is received by the device factory core. Therefore, if you need to be stateful, you should keep track of the state based on the payload emitter.
+* a parser must be thread safe: the device factory core might called concurrently by different threads.
+
+## Parser interface
+
+A device factory parser is registered as a service implementing the `org.eclipse.sensinact.gateway.southbound.device.factory.IDeviceMappingParser` interface and associated to two properties:
+* `sensinact.southbound.mapping.parser`: the ID of the parser. The constant is defined by `org.eclipse.sensinact.gateway.southbound.device.factory.IDeviceMappingParser.PARSER_ID`
+* `sensinact.southbound.mapping.types`: a list of MIME types supported by the parser. This is used if a payload is received with a content type but no parser ID has been configured for that entry. If can be set to `null` to only accept payloads explicitly associated to the parser. The constant is defined by `org.eclipse.sensinact.gateway.southbound.device.factory.IDeviceMappingParser.PARSER_SUPPORTED_TYPES`.
+
+This `IDeviceMappingParser` interface defines only one method:
+* ```java
+  List<? extends IDeviceMappingRecord> parseRecords(byte[] rawInput, Map<String, Object> parserConfiguration, Map<String, String> context) throws ParserException
+  ```
+
+  It accepts the following arguments:
+  * `rawInput`: the raw payload as given to the device factory core. Note that the transport that received it might have done some pre-treatment.
```

Comment:
```suggestion
  * `rawInput`: the raw payload as given to the device factory core. Note that the transport that received it might have done some pre-treatment, for example unzipping data that was compressed for transmission.
```